openedx/core/djangoapps/course_apps/models.py

```diff
@@ -1,3 +1,63 @@
 """
-Models for course apps.
+Models to store course apps data.
 """
+from typing import Dict
+
+from django.db import models
+from model_utils.models import TimeStampedModel
+from opaque_keys.edx.django.models import CourseKeyField
+from opaque_keys.edx.keys import CourseKey
+from openedx.core.djangoapps.course_apps.plugins import CourseApp
+from simple_history.models import HistoricalRecords
+
+
+class CourseAppStatus(TimeStampedModel):
+    """
+    CourseAppStatus stores the enabled/disabled status for course apps.
+
+    .. no_pii:
+    """
+
+    app_id = models.CharField(max_length=32, db_index=True)
+    enabled = models.BooleanField(default=False)
+    course_key = CourseKeyField(max_length=255, db_index=True)
+
+    history = HistoricalRecords()
+
+    @classmethod
+    def get_all_app_status_data_for_course(cls, course_key: CourseKey) -> Dict[str, bool]:
+        """
+        Get a dictionary containing the status of all apps linked to the course.
+
+        Args:
+            course_key (CourseKey): the course id for which app status is needed
+
+        Returns:
+             A dictionary where the keys are app ids and the values are booleans
+             indicating if the app with that id is enabeld
+        """
+        return dict(
+            cls.objects.filter(course_key=course_key).values("app_id", "enabled")
+        )
+
+    @classmethod
+    def update_status_for_course_app(cls, course_key: CourseKey, course_app: CourseApp):
+        """
+        Creates or updates a status entry for the specified app and course.
+
+        Args:
+            course_key (CourseKey): the course id for which the app status is to
+                be set
+            course_app (CourseApp): the course app object which needs to be updated
+        """
+        is_enabled = course_app.is_enabled(course_key=course_key)
```

Comment:
I don't think we need to couple this with `course_app.is_enabled`. This method should simply know how to update status of an app not _fetch_ and update status of an app.

Reply:
I'm a bit confused here, how does that help then? 

We already have a method (update_or_create) that takes the course ID, app ID and enabled and saves that, so then this function would just pass on exactly the same parameters to another method under the same class. How does this simplify things?

Reply:
Thanks for this question, I can explain.

The method is intended such that objects outside the data model don't need to worry about its data struct. The data model is aware/responsible for storing/updating its objects. It knows which keys to map "course_key, app and status". 

It's not a blocker for me, its just cleans up the code, so I am fine if you disagree :)